### PR TITLE
Pass correct event target from the bookend to the AMP linker

### LINKED ISSUE #23260: AMP linker not working with Stories bookend links.
AMP linker not working with Stories bookend links.

To repro:

1. Navigate to [this story](https://www.laroche-posay.fr/article/anthelios-story/amp-a36900.aspx)
2. Navigate to the last page of the story and click on the CTA.
3. GOOD: notice the `_gl` GET parameter that has been added to the URL by the AMP linker
4. Click on the link from the bookend
5. BAD: notice how this parameter has not been added

### PR DESCRIPTION
**Pull request by Enriqe**
fixes #23260

When clicking a link from the bookend, the URL wasn't being decorated with the `_gl` parameter because the bookend is inside a shadow DOM. 

This is because whenever the `handle_` method in `navigation.js` received the event, the `target` would correspond to the entire shadow DOM root, instead of the actual clicked target.

This PR allows custom AMP targets to be forwarded on the event by using a custom property, then `navigation.js` can check if this property exists, and use that instead of the default `target` that comes in the event.

ezgif-1-de302fb497d8

### CODE REVIEW

File: src/service/navigation.js [hunk 1]
```diff
@@ -43,6 +43,9 @@ const VALID_TARGETS = ['_top', '_blank'];
 /** @private @const {string} */
 const ORIG_HREF_ATTRIBUTE = 'data-a4a-orig-href';
 
+/** @const {string} */
+export const AMP_CUSTOM_LINKER_TARGET = '__AMP_CUSTOM_LINKER_TARGET__';
```
Comment (gmajoulet) [inline src/service/navigation.js]: Please add a comment to explain that this is used to retarget events originating from shadow DOM

File: extensions/amp-story/1.0/bookend/amp-story-bookend.js [hunk 2]
```diff
@@ -14,6 +14,7 @@
  * limitations under the License.
  */
 
+import {AMP_CUSTOM_LINKER_TARGET} from '../../../../src/service/navigation';
```
Comment (gmajoulet) [inline extensions/amp-story/1.0/bookend/amp-story-bookend.js]: Services can be loaded asynchronously and should not be bundled in the extension code. It'd just add this service code into the amp-story extension, even though it's already present in the v0 file.
You could either have a getter in the navigation service, or just duplicate the constant

**Comment by Enriqe** [comment]
Ready for another review 👍

File: test/unit/test-navigation.js [hunk 3]
```diff
@@ -118,6 +124,17 @@ describes.sandboxed('Navigation', {}, () => {
           expect(handleCustomProtocolSpy).to.be.calledWith(event, anchor);
         });
 
+        it.only('should select a custom linker target', () => {
```
Comment (alabiaga) [inline test/unit/test-navigation.js]: remove only

Comment (Enriqe) [reply test/unit/test-navigation.js]: woops, thanks for noticing

File: test/unit/test-navigation.js [hunk 4]
```diff
@@ -118,6 +124,17 @@ describes.sandboxed('Navigation', {}, () => {
           expect(handleCustomProtocolSpy).to.be.calledWith(event, anchor);
         });
 
+        it('should select a custom linker target', () => {
+          const handleClickSpy = sandbox.spy(handler, 'handleClick_');
```
Comment (gmajoulet) [inline test/unit/test-navigation.js]: I'm not familiar with this file, but both stubbing a private method and manually calling a private method are considered bad unit test patterns. They're testing the actual implementation and not the behavior, which reduces the tests coverage and makes the code harder to refactor.

Is there any way to improve this at this test level? Of course you don't have to refactor the whole file, I'm just wondering if there's a way to improve this `sandbox.spy`, not `handler.handle_` as it seems to be used across the file.
(Still LGTM)

Comment (Enriqe) [reply test/unit/test-navigation.js]: That's true, it's just the way I found to testing this type of behavior on this file. 

I changed it to use the previously defined `handleNavSpy` so I don't create another one like this, which is a little bit better.

Comment (alabiaga) [reply test/unit/test-navigation.js]: Good point. I think we can even just change this particular test to invoke the handler via the document used to get the navigation service. e.g document.click(). Add a TODO to fix this throughout the test.

Comment (Enriqe) [reply test/unit/test-navigation.js]: Sg, done.

**Pull request merged**